src/safe_cli/safe_cli.py

```diff
@@ -0,0 +1,128 @@
+        """
+        self.safe_address = safe_address
+        self.node_url = node_url
+        if history:
+            self.session = PromptSession(
+                history=FileHistory(os.path.join(sys.path[0], ".history"))
+            )
+        else:
+            self.session = PromptSession()
+        self.safe_operator = SafeOperator(safe_address, node_url)
+        self.prompt_parser = PromptParser(self.safe_operator)
+
+    def print_startup_info(self):
+        print_formatted_text(text2art("Safe CLI"))  # Print fancy text
+        print_formatted_text(HTML(f"<b>Version: {VERSION}</b>"))
+        print_formatted_text(
+            HTML("<b><ansigreen>Loading Safe information...</ansigreen></b>")
+        )
+        self.safe_operator.print_info()
+
+        print_formatted_text(
+            HTML("\nUse the <b>tab key</b> to show options in interactive mode.")
+        )
+        print_formatted_text(
+            HTML(
+                "The <b>help</b> command displays all available options and the <b>exit</b> command terminates the safe-cli."
+            )
+        )
+
+    def get_prompt_text(self):
+        mode: Optional[str] = "blockchain"
+        if isinstance(self.prompt_parser.safe_operator, SafeTxServiceOperator):
+            mode = "tx-service"
+
+        return HTML(
+            f"<bold><ansiblue>{mode} > {self.safe_address}</ansiblue><ansired> > </ansired></bold>"
+        )
+
+    def get_bottom_toolbar(self):
+        return HTML(
+            f'<b><style fg="ansiyellow">network={self.safe_operator.network.name} '
+            f"{self.safe_operator.safe_cli_info}</style></b>"
+        )
+
+    def parse_operator_mode(self, command: str) -> Optional[SafeOperator]:
+        """
+        Parse operator mode to switch between blockchain (default) and tx-service
+        :param command:
+        :return: SafeOperator if detected
+        """
+        split_command = command.split()
+        try:
+            if (split_command[0]) == "tx-service":
+                print_formatted_text(
+                    HTML("<b><ansigreen>Sending txs to tx service</ansigreen></b>")
+                )
+                return SafeTxServiceOperator(self.safe_address, self.node_url)
+            elif split_command[0] == "blockchain":
+                print_formatted_text(
+                    HTML("<b><ansigreen>Sending txs to blockchain</ansigreen></b>")
+                )
+                return self.safe_operator
+        except SafeServiceNotAvailable:
+            print_formatted_text(
+                HTML("<b><ansired>Mode not supported on this network</ansired></b>")
+            )
+
+    def get_command(self) -> str:
+        return self.session.prompt(
+            self.get_prompt_text,
+            auto_suggest=AutoSuggestFromHistory(),
+            bottom_toolbar=self.get_bottom_toolbar,
+            lexer=PygmentsLexer(SafeLexer),
+            completer=SafeCompleter(),
+        )
+
+    def loop(self):
```

Comment:
When starts "print_startup_info" method, a message is being displayed that shows:

"Version: 1.2.1
Loading Safe information..." 

and then waits for the response with the Safe information. Do you think we should change something here?

Reply:
As discussed, it seems that there is no delay except in one situation: if `--get-safes-from-owner` is enabled then we need to remotely connect to another service which can take some time.

We should consider providing some type of feedback to the user where we indicate that it's fetching data (this happens before the call to `print_startup_info` https://github.com/safe-global/safe-cli/blob/ded64e7a800979f5394137648871f3376d67b269/src/safe_cli/main.py#L328

It can be done on a separate PR if we decide doing this 👍

Reply:
Done [here](https://github.com/safe-global/safe-cli/pull/419/commits/7f61716537adc770043c238b88587ea86260729f)